pkg/controller/worker/machines.go

```diff
@@ -177,6 +177,15 @@ func (w *workerDelegate) generateMachineConfig() error {
 				}
 			}
 
+			if workerConfig.NodeTemplate != nil {
```

Comment:
This can be better written as if..elseif
```
if workerConfigNodeTemplate is provided{
                  // use it
} else if poolNodeTemplate is provided{
                  // use it
}
```